ERCS/erc-7777.md

```diff
@@ -0,0 +1,137 @@
+---
+eip: 7777
+title: Common Quote Oracle
+description: Standard API for data feeds providing the relative value of assets in the Ethereum and EVM-compatible blockchains.
+author: alcueca (@alcueca), ruvaag (@ruvaag), totomanov (@totomanov), r0ohafza (@r0ohafza)
+discussions-to: https://ethereum-magicians.org/t/erc-for-oracle-value-feeds/20351
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-06-20
+---
+
+## Abstract
+
+The following standard allows for the implementation of a standard API for data feeds providing the relative value of
+assets in the Ethereum and EVM-compatible blockchains.
+
+## Motivation
+
+The information required to value assets is scattered over a number of major and minor sources, each one with their own
+integration API and security considerations. Many protocols over the years have implemented oracle adapter layers for
+their own use to abstract this complexity away from their core implementations, leading to much duplicated effort.
+
+This specification provides a standard API aimed to serve the majority of use cases. Preference is given to ease of
+integration and serving the needs of product teams with less knowledge, requirements and resources.
+
+## Specification
+
+### Definitions
+
+- base asset: The asset that the user needs to know the value for (e.g: USDC as in "I need to know the value of 1e6 USDC
+  in ETH terms").
+- quote asset: The asset in which the user needs to value the `base` (e.g: ETH as in "I need to know the value of 1e6
+  USDC in ETH terms").
+- value: An amount of `base` in `quote` terms (e.g. The `value` of 1000e6 USDC in ETH terms is 283,969,794,427,307,000
+  ETH, and the `value` of 1000e18 ETH in USDC terms is 3,521,501,299,000 USDC). Note that this is an asset amount, and
+  not a decimal factor.
```

Comment:
Here, there is a risk of uint256 overflow. I think that a note about it would be useful, ideally with some suggestion about what to throw if that happens.

Reply:
For example, the maximum accepted value for `base` may be the total supply.

Reply:
There is certainly a risk of reverting, but I don't think that there is a reason to prescribe the error that should be thrown, I don't see a problem of each implementation reverts with a different error. The error won't be frequent enough to warrant the effort, and if we mandate something here it might add complexity.

Likewise, I wouldn't ask for a total supply check. The protection offered doesn't justify the extra gas imho. Consider that even if we limit `getQuote` to the total supply, that still doesn't mean that the total supply could be exchanged due to liquidity restraints.

Reply:
I added a condition to make sure people know to revert on overflows, even if it is a bit obvious.